### ISSUE TITLE
[Bug][UITests] Save collection fails intermittently - needs dev investigation

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#10587]
## Steps to reproduce
Run the tests from CollectionTest.kt on Firebase

### Expected behavior
Collections are created in automation tests.

### Actual behavior
While running the tests, sometimes the collections are not created or tabs not added to the existing collection.
The confirmation snack bar doesn't appear either.

### Device information
* Android device: Virtual device: Pixel 2 (API 28), x86
* Fenix version: geckoNightlyDebug
* Cannot reproduce manually on an emulator or physical device.

Examples:
https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8016584804846223411/executions/bs.b2925233b2591f58/testcases/1/test-cases

https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8016584804846223411/executions/bs.c2bb277ecf2e20bd/testcases/1/test-cases

https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8016584804846223411/executions/bs.d40df2f07cb64f33/testcases/1/test-cases

https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/6416467061915155594/executions/bs.26645344d9bb048a/videos

On the same note: editBookmark changes are sometimes not saved either (maybe the same cause?)
see: https://github.com/mozilla-mobile/fenix/issues/10690

**Comment by sv-ohorvath** [comment]
It actually seems that the collection is not getting saved sometimes.

**Comment by sv-ohorvath** [comment]
cc: @mcarare

**Comment by rpappalax** [comment]
Waiting on feedback from Firebase#test-lab

**Comment by sv-ohorvath** [comment]
> https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/8016584804846223411/executions/bs.b2925233b2591f58/testcases/1/test-cases

I was able to reproduce adding a tab to an existing collection and not getting it saved (no snackbar either), both manually and while running the test automatically, but only on the emulator. 
After I ran the tests automatically, I was left with the last collection created.
After that, I don't have some clear steps:
1. I opened wikipedia and added it to the collection.
2. Then I just removed a tab from the collection, go back to that tab and try adding it again.

Unfortunately, I can't reproduce the other cases where you create a new collection the first time.

@mcarare Maybe this might be a hint, so if you could have another look at this, try with the following: running the whole CollectionTest class, the emulator has wifi off, running on Data connection, on LTE, with poor connection speed. 

@rpappalax @AaronMT I think we can postpone a little while asking the test-lab community until we can have a clear response from our side, this looks suspicious. If mcarare still can't reproduce maybe we can ask someone else to give it a try.

**Comment by mcarare** [comment]
@sv-ohorvath I am not able to reproduce this. Also, you can save tabs to collections without even being connected to internet, I do not think network settings on emulator matter in reproducing this. It might have to do with the saving to database, but the code for that is straightforward, I do not see any reason saving a tab should fail.

**Comment by sv-ohorvath** [comment]
The collections flow was changed and I think there are some more changes on the way, so this issue might be gone already. Leaving the issue opened to be re-written and re-enabled when possible with the new flow.
